### ISSUE TITLE
feature request: autofill for Android Oreo

### ISSUE THREAD

**Issue by Wohlraj** [turtl/mobile#37]
As the title says.
See Android 8 autofill API.

**Comment by orthecreedence** [comment]
Cool idea. I'll see what I can do.

**Comment by thedward** [comment]
This appears to be working now. When Oreo was first released Chrome (and thus webviews) did not support the new autofill service. I'm not sure when Chrome added support, but it was recently.

- Android — 8.10
- Chrome — 65.0.3325 109
- Turtl — 0.6.4

**Comment by Wohlraj** [comment]
What has Chrome got to do with this?
It is not installed in my phone, and autofill works for many apps, but not Turtl.

**Comment by thedward** [comment]
@Wohlraj, Turtl for Android is built with [Cordova](https://cordova.apache.org) — which means it it basically HTML & JavaScript that acts like a native app, but is rendered by the system webview (basically a web browser as a component). 

If you have Chrome installed, it will override the webview that is built into the base system (Android System Webview) and thus can change the behavior of apps that use webview (like Turtl).

If the webview Turtl is using doesn't support the autofill API, then Turtl isn't going to support it. Cordova enables only very limited access to system APIs.

Maybe you could try installing Chrome?

**Comment by Wohlraj** [comment]
Thank you for explaining. I installed Chrome, but still did not see a prompt for KeePass password manager. Maybe we are taking about two different things?

**Comment by thedward** [comment]
@Wohlraj  

No. I was simply mistaken. Chrome and webviews still don't support the autofill API — it turns out that my password manager uses the autofill API when possible, but falls back to a work around method using the accessibility API for apps that do not support autofill.  

Hopefully autofill support will be added for webviews and browsers in the near future.
